dcos/http.py

```diff
@@ -414,7 +428,7 @@ def _get_dcos_acs_auth(username, password, hostname):
 
         # using private method here, so we don't retry on this request
         # error here will be bubbled up to _request_with_auth
-        response = _request('post', url, json=creds)
+        response = _request('post', url, json=creds, verify=_verify_ssl())
```

Comment:
There is side effect in `_verify_ssl`. This style of coding is hard to debug and understand in the future. How can we fix this?